lib/asciidoctor/extensions.rb

```diff
@@ -1481,17 +1481,36 @@ def resolve_class object
     # Public: Resolves the Class object for the qualified name.
     #
     # Returns Class
-    def class_for_name qualified_name
-      resolved = ::Object
-      (qualified_name.split '::').each do |name|
-        unless name.empty? || ((resolved.const_defined? name) && ::Module === (resolved = resolved.const_get name))
-          raise ::NameError, %(Could not resolve class for name: #{qualified_name})
+    if RUBY_MIN_VERSION_2
+      def class_for_name qualified_name
+        resolved = ::Object.const_get qualified_name, false
+        raise unless ::Class === resolved
+        resolved
+      rescue
+        raise ::NameError, %(Could not resolve class for name: #{qualified_name})
+      end
+    elsif RUBY_MIN_VERSION_1_9
```

Comment:
Ruby 1.9 is not EOL yet ?

Reply:
Yes. But we need a dedicated release and announcement before we can drop support. Until then, I have to keep the test suite running as is.